src/shared/pager.c

```diff
@@ -147,10 +150,16 @@ int pager_open(bool no_pager, bool jump_to_end) {
         }
 
         /* Return in the parent */
-        if (dup2(fd[1], STDOUT_FILENO) < 0)
+        stored_stdout = dup(STDOUT_FILENO);
+        if (dup2(fd[1], STDOUT_FILENO) < 0) {
```

Comment:
i figure i sympathize to a certain level with this change, it probably does make sense to restore the original stdout/stderr when we close the pager...

However, I am pretty sure, the copies of the original fds should carry O_CLOEXEC (unlike the originals). Moreover, I am pretty sure this should use fcntl(…, FD_DUP_CLOEXEC, 3) instead of plain dup(), so that we never duplicate into fd 0, 1, 2 should they be closed...